build/Dockerfile

```diff
@@ -87,6 +87,43 @@ RUN mkdir build src && cd build \
 
 RUN apt-get autoremove -y gcc g++
 
+# Build & install libtinfo for the linux targets.
+# (on BSD or BSD-derived like macOS it's already built-in; on windows we don't need it.)
+
+# Configure flags.
+COPY ncurses.conf ./
+# The patch is needed to work around a bug in Debian mawk, see
+# http://lists.gnu.org/archive/html/bug-ncurses/2015-08/msg00008.html
+COPY ncurses.patch ./
+
+# Run the two builds.
+# We need to manually disable secure_getenv and cdefs on the musl build
+# because they are not available in the musl sysroot.
+RUN mkdir ncurses \
+ && curl -fsSL http://ftp.gnu.org/gnu/ncurses/ncurses-6.0.tar.gz | tar --strip-components=1 -C ncurses -xz \
+ && cd ncurses \
+ && patch -p0 <../ncurses.patch \
+ && export cf_cv_type_of_bool='unsigned char' \
+ && export cf_cv_working_poll=yes \
```

Comment:
This looks suspicious. It deserves a comment at least, I think?

Reply:
Done.